go-controller/pkg/ovn/policy.go

```diff
@@ -316,6 +366,116 @@ func (oc *Controller) createDefaultDenyPortGroup(policyType knet.PolicyType) {
 	}
 }
 
+// Creates a policy to allow multicast traffic within 'ns':
+// - a port group containing all logical ports associated with 'ns'
+// - one "from-lport" ACL allowing egress multicast traffic from the pods
+//   in 'ns'
+// - one "to-lport" ACL allowing ingress multicast traffic to pods in 'ns'.
+//   This matches only traffic originated by pods in 'ns' (based on the
+//   namespace address set).
+func (oc *Controller) createMulticastAllowPolicy(ns string) {
+
+	portGroupName := "mcastPortGroup-" + ns
+	portGroupHash := hashedPortGroup(portGroupName)
+	portGroupUUID, err := oc.createPortGroup(portGroupName, portGroupHash)
+	if err != nil {
+		logrus.Errorf("Failed to create port_group for %s (%v)",
+			portGroupName, err)
+		return
+	}
+
+	err = oc.addACLPortGroup(portGroupUUID, portGroupHash, fromLport,
+		defaultMcastAllowPriority, "ip4.mcast", "allow",
+		knet.PolicyTypeEgress)
+	if err != nil {
+		logrus.Errorf("Failed to create allow egress multicast ACL for %s (%v)",
+			ns, err)
+		return
+	}
+
+	nsAddressSet := hashedAddressSet(ns)
+	match := fmt.Sprintf("ip4.src == $%s && ip4.mcast", nsAddressSet)
+	err = oc.addACLPortGroup(portGroupUUID, portGroupHash, toLport,
+		defaultMcastAllowPriority, match, "allow", knet.PolicyTypeIngress)
+	if err != nil {
+		logrus.Errorf("Failed to create allow ingress multicast ACL for %s (%v)",
+			ns, err)
+		return
+	}
+
+	// Add all ports from this namespace to the multicast allow group.
+	for _, portName := range oc.namespaceAddressSet[ns] {
+		oc.addToACL(portGroupHash, portName)
+	}
+}
+
+// Delete the policy to allow multicast traffic within 'ns'.
+func (oc *Controller) deleteMulticastAllowPolicy(ns string) {
+
+	portGroupName := "mcastPortGroup-" + ns
+	portGroupHash := hashedPortGroup(portGroupName)
+
+	err := oc.deleteACLPortGroup(portGroupHash, fromLport,
+		defaultMcastAllowPriority, "ip4.mcast", "allow",
+		knet.PolicyTypeEgress)
+	if err != nil {
+		logrus.Errorf("Failed to delete allow egress multicast ACL for %s (%v)",
+			ns, err)
+		return
+	}
+
+	nsAddressSet := hashedAddressSet(ns)
+	match := fmt.Sprintf("ip4.src == $%s && ip4.mcast", nsAddressSet)
+	err = oc.deleteACLPortGroup(portGroupHash, toLport,
+		defaultMcastAllowPriority, match, "allow", knet.PolicyTypeIngress)
+	if err != nil {
+		logrus.Errorf("Failed to delete allow ingress multicast ACL for %s (%v)",
+			ns, err)
+		return
+	}
+
+	oc.deletePortGroup(portGroupHash)
+}
+
+// Creates a global default deny multicast policy:
+// - one ACL dropping egress multicast traffic from all ports in group
+//   'ingressDefaultDeny': this is to protect OVN controller from processing
+//   IP multicast reports from nodes that are not allowed to receive multicast
+//   traffic.
+// - one ACL dropping ingress multicast traffic from all ports in group
+//   'ingressDefaultDeny'
+func (oc *Controller) createDefaultDenyMulticastPolicy() {
```

Comment:
@dceara same here, let's return errors and make the caller log

Reply:
Done.